test/functional/terminal/tui_spec.lua

```diff
@@ -1529,119 +1529,44 @@ describe("TUI", function()
 end)
 
 describe('TUI bg color', function()
-  local screen
-
-  local function setup()
-    -- Only single integration test.
-    -- See test/unit/tui_spec.lua for unit tests.
-    clear()
-    screen = thelpers.screen_setup(0, '["'..nvim_prog
-      ..'", "-u", "NONE", "-i", "NONE", "--cmd", "set noswapfile", '
-      ..'"-c", "autocmd OptionSet background echo \\"did OptionSet, yay!\\""]')
-  end
-
-  before_each(setup)
-
-  it('triggers OptionSet event on unsplit terminal-response', function()
-    screen:expect([[
-    {1: }                                                 |
-    {4:~                                                 }|
-    {4:~                                                 }|
-    {4:~                                                 }|
-    {5:[No Name]                       0,0-1          All}|
-                                                      |
-    {3:-- TERMINAL --}                                    |
-    ]])
-    feed_data('\027]11;rgb:ffff/ffff/ffff\007')
-    screen:expect{any='did OptionSet, yay!'}
-
-    feed_data(':echo "new_bg=".&background\n')
-    screen:expect{any='new_bg=light'}
-
-    setup()
-    screen:expect([[
-    {1: }                                                 |
-    {4:~                                                 }|
-    {4:~                                                 }|
-    {4:~                                                 }|
-    {5:[No Name]                       0,0-1          All}|
-                                                      |
-    {3:-- TERMINAL --}                                    |
-    ]])
-    feed_data('\027]11;rgba:ffff/ffff/ffff/8000\027\\')
-    screen:expect{any='did OptionSet, yay!'}
+  before_each(clear)
 
-    feed_data(':echo "new_bg=".&background\n')
-    screen:expect{any='new_bg=light'}
+  it('is properly set in a nested Neovim instance when background=dark', function()
+    command('set background=dark')  -- set outer Neovim background
+    local screen = thelpers.screen_setup(0, string.format(
+      '["%s", "-u", "NONE", "-i", "NONE", "--cmd", "set noswapfile"]', nvim_prog))
+    retry(nil, 30000, function()  -- wait for automatic background processing
+      screen:sleep(20)
+      feed_command('set background?')  -- check nested Neovim background
+      screen:expect([[
+      {1: }                                                 |
+      {4:~                                                 }|
+      {4:~                                                 }|
+      {4:~                                                 }|
+      {5:[No Name]                       0,0-1          All}|
+        background=dark                                 |
+      {3:-- TERMINAL --}                                    |
+      ]])
+    end)
   end)
 
-  it('triggers OptionSet event with split terminal-response', function()
```

Comment:
why isn't OptionSet being tested now? is it no longer triggered?

Reply:
That test was starting a nested Nvim instance. On startup, that nested instance would query for background color, but there would be no response. The test would then simulate a background color response and check that Nvim handled accordingly by setting a background option.

Now when the nested instance queries for background color there is a response, so a simulated response after startup doesn't work in this context since Nvim is no longer in a state to handle the response.

Reply:
We still want to test that the OptionSet event fires though. Can you include that somewhere in the new tests?

Reply:
Done (29b04019eb27ddb417c737979750c571ace71a02).